### PR TITLE
Improvements/adding additional checks

### PR DESCRIPTION
**Pull request by keichiri**

### CODE REVIEW

File: libindy/src/services/wallet/storage/default/mod.rs [hunk 1]
```diff
@@ -176,7 +177,13 @@ impl StorageIterator for SQLiteStorageIterator {
                     None
                 };
                 let tags = if self.options.fetch_tags {
-                    Some(self.tag_retriever.as_mut().unwrap().retrieve(row.get(0))?)
+                    if let Some(ref mut tag_retriever) = &mut self.tag_retriever {
```
Comment (Artemkaaas) [inline libindy/src/services/wallet/storage/default/mod.rs]: I have compilation error here. It looks like you use another version of Rust

File: libindy/src/errors/wallet.rs [hunk 2]
```diff
@@ -278,19 +294,22 @@ pub enum WalletQueryError {
     ParsingErr(String),
     StructureErr(String),
     ValueErr(String),
+    CommonError(CommonError),
 }
```
[the remark this answers is not in the document]
Comment (keichiri) [reply libindy/src/errors/wallet.rs]: Oh yes, I first used it but later changed approach and forgot to remove. You are right

File: libindy/src/services/wallet/language.rs [hunk 3]
```diff
@@ -89,6 +89,35 @@ impl Operator {
         }
```
[the remark this answers is not in the document]
Comment (keichiri) [reply libindy/src/services/wallet/language.rs]: yes its not necessary now you are correct

File: libindy/src/services/wallet/storage/default/mod.rs (hunk 1 above)
Comment (keichiri) [reply libindy/src/services/wallet/storage/default/mod.rs]: interesting. Yes, the syntax looks a bit strange, maybe it was not possible before. I use 1.26

**Pull request merged**